### ISSUE TITLE
Updating Touchables for basic accessibility in rnwcpp

### ISSUE THREAD

**Issue by ahimberg** [microsoft/react-native-windows#2088]
keyboard, onaccessibilitytap support in addition to focus (already done)

[yulikl] I believe Touchables should behave exactly like Buttons from an accessibility perspective (well, possibly with the exception of TouchableWithoutFeedback).

**Comment by MikeHillberg** [comment]
Do I understand onaccessibilitytap correctly that it's basically a double-tap event and has nothing to do with accessibility?

**Comment by ahimberg** [comment]
I meant something along the lines of firing an event in response to a button role automationpeer's invoke.  We have onaccessibilitytap as a name used already by the windows C# version, not sure we'd use that exact name.  https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.buttonautomationpeer.invoke

**Comment by MikeHillberg** [comment]
When the automation peer is invoked, doesn't that just raise the button's onPress?

**Comment by ahimberg** [comment]
This is mostly fixed at this point.  when using touchables, if you set its accessibleRole='button' it will now appear as a button and support accessibility tool invoke.   Maybe we need to do a little more so that you don't have to set this property but that is pretty consistent with other RN platforms

**Comment by YuliKl** [comment]
Is it possible to set accessibleRole by default, without requiring that every app remembers to do this? If the accessibleRole for every touchable can be set, I think we should use this issue to track the remaining work to set this by default.

**Comment by kmelmon** [comment]
We think what's required here is to create a touchabeWindows and eventually push core changes back to react-native, possibly into Pressable when that is created.

**Comment by harinikmsft** [comment]
P0 needs:
- Need to add more accessibility roles for list, listitem
- Need to add support for accessibilityStates
- Need to expose accessibility roles for "x of n count" in lists

P1 needs: 
- see if we need to add a default for some of the JS components for these roles.

**Comment by kmelmon** [comment]
consolidating accessibility tasks - Jon will be taking these

**Comment by jonthysell** [comment]
Some accessibilityStates already exist, just need to add the remaining. List and listitem roles need to be invented JS side and then plumbed through, then mapped to the right automationcontroltype.

**Comment by jonthysell** [comment]
Closing this, it's now covered by #2578, #2579
